File: app/src/main/java/com/phpexpert/bringme/models/LoginViewModel.java
package com.phpexpert.bringme.models;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class LoginViewModel extends ViewModel {

    private final MutableLiveData<Event<Boolean>> backPressed = new MutableLiveData<>();
    public void backPressed() {
        backPressed.setValue(new Event<>(true));
    }
    public LiveData<Event<Boolean>> getBackPressedEvent() {
        return backPressed;
    }

    private final MutableLiveData<Event<Boolean>> openLoginScreen = new MutableLiveData<>();
    public void setContinueWithFragment() {
        openLoginScreen.setValue(new Event<>(true));
    }
    public LiveData<Event<Boolean>> getContinueWithFragment() {
        return openLoginScreen;
    }


    private final MutableLiveData<Event<Object[]>> openOtpFragment = new MutableLiveData<>();
    public void setOpenVerifyOtpfragment(Object[] objects) {
        openOtpFragment.setValue(new Event<>(objects));
    }
    public LiveData<Event<Object[]>> getOpenVerifyOtpScreen() {
        return openOtpFragment;
    }


    private final MutableLiveData<Event<Boolean>> phoneVarificationFrag = new MutableLiveData<>();
    public void setPhoneVarificationFrag() {
        phoneVarificationFrag.setValue(new Event<>(true));
    }
    public LiveData<Event<Boolean>> getPhoneVarificationFrag() {
        return phoneVarificationFrag;
    }
    private final MutableLiveData<Event<Boolean>> registerFrag = new MutableLiveData<>();
    public void setRegisterFrag() {
        registerFrag.setValue(new Event<>(true));
    }
    public LiveData<Event<Boolean>> getRegisterFrag() {
        return registerFrag;
    }






}

File: app/src/main/java/com/phpexpert/bringme/models/LoginViewModel.kt
package com.phpexpert.bringme.models

import android.content.Context
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.phpexpert.bringme.dtos.ForgotPasswordChangeDtoMain
import com.phpexpert.bringme.dtos.ForgotPasswordDtoMain
import com.phpexpert.bringme.dtos.LoginDtoMain
import com.phpexpert.bringme.dtos.ResendOtpMain
import com.phpexpert.bringme.repositories.LoginRepo

class LoginViewModel : ViewModel() {

    private var loginData = MutableLiveData<LoginDtoMain>()
    private var otpForgotPasswordSend = MutableLiveData<ForgotPasswordDtoMain>()
    private var otpForgotPasswordReset = MutableLiveData<ForgotPasswordChangeDtoMain>()
    private var loginDetailsData = MutableLiveData<ResendOtpMain>()


    fun getLoginData(mapData: Map<String, String>): LiveData<LoginDtoMain> {
        loginData = LoginRepo().getLoginData(mapData)
        return loginData
    }

    fun getOtpForgotPasswordSendData(mapData: Map<String, String>): LiveData<ForgotPasswordDtoMain> {
        otpForgotPasswordSend = LoginRepo().getForgotPasswordOtpSend(mapData)
        return otpForgotPasswordSend
    }

    fun getOtpForgotPasswordReset(mapData: Map<String, String>): LiveData<ForgotPasswordChangeDtoMain> {
        otpForgotPasswordReset = LoginRepo().getForgotPasswordReset(mapData)
        return otpForgotPasswordReset
    }
    fun getLoginDetailsData(mapData: Map<String, String>): LiveData<ResendOtpMain> {
        loginDetailsData = LoginRepo().getLoginDetailsData(mapData)
        return loginDetailsData
    }

}